File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/user/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node): # let's us inherit everything from Node
    def __init__(self):
        super().__init__("py_test")
        self.counter_ = 0
        self.get_logger().info("What's up!!")
        self.create_timer(1.0, self.timer_callback) # every 1 second call timer call back function

    def timer_callback(self):
        self.get_logger().info("Hiiiiii" + str(self.counter_))
        self.get_logger().info(f"Weeeee {self.counter_}")
        self.counter_ += 1

def main(args=None):
    rclpy.init(args=args) # initialize ros2
    node = MyNode() # create node from constructor
    rclpy.spin(node) # keeps node alive until i press ctrl C
    rclpy.shutdown() # shutdown node

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node // inherit from rclcpp::Node
{
public:
    MyNode() : Node("cpp_test"), counter_(0) // build constructor for parent class and initialize counter
    {
        RCLCPP_INFO(this->get_logger(), ":))))))");
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                        std::bind(&MyNode::timerCallBack, this));
    }
private:
    void timerCallBack()
    {
        RCLCPP_INFO(this->get_logger(), ":ppppppp");
        counter_++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv); // initiazlize ros2 communications
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}